### PR TITLE
chore(tracing): add `db.system` tag to mongodb integrations

### PR DESCRIPTION
**Pull request by jbertran**
## Description

Part of the effort to unify tag names across integrations and languages involve setting the `db.system` tag when tracing DB and KV store integrations.

This PR sets the tag on all mongodb integrations:
* `pymongo`
* `mongoengine`

All integration tests that check tags now _also_ check for the `db.system` tag. Those that did not (because the integration tested inherits its behaviour from another) have no test changes.

All relevant snapshots are updated with the addition of the new expected tag.

## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/contributing.html#Release-Note-Guidelines) are followed.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.

### CODE REVIEW

File: ddtrace/contrib/pymongo/patch.py [hunk 1]
```diff
@@ -10,16 +10,15 @@
...
 from .client import TracedMongoClient
 from .client import set_address_tags
 
 
 config._add(
     "pymongo",
-    dict(
-        _default_service="pymongo",
-    ),
+    dict(_default_service="pymongo", dbms_name="mongodb"),
```
Comment (mabdinur) [inline ddtrace/contrib/pymongo/patch.py:21]: Is this the dbms_name static or configurable. If it's static it should not live in the config object. We can just define and use a constant. 
```suggestion
    dict(
        _default_service="pymongo",
    ),
```

File: ddtrace/contrib/pymongo/client.py [hunk 2]
```diff
@@ -219,6 +221,7 @@ def __trace(self, cmd):
         s = pin.tracer.trace("pymongo.cmd", span_type=SpanTypes.MONGODB, service=pin.service)
 
         s.set_tag_str(COMPONENT, config.pymongo.integration_name)
+        s.set_tag_str(db.SYSTEM, config.pymongo.dbms_name)
```
Comment (mabdinur) [inline ddtrace/contrib/pymongo/client.py:224]:
```suggestion
        s.set_tag_str(db.SYSTEM, mongox.SERVICE)
```

**Review by emmettbutler — DISMISSED** [review]

File: ddtrace/contrib/pymongo/patch.py [hunk 3]
```diff
@@ -71,6 +70,7 @@ def traced_get_socket(wrapped, instance, args, kwargs):
         "pymongo.get_socket", service=trace_utils.int_service(pin, config.pymongo), span_type=SpanTypes.MONGODB
     ) as span:
         span.set_tag_str(COMPONENT, config.pymongo.integration_name)
+        span.set_tag_str(db.SYSTEM, config.pymongo.dbms_name)
```
Comment (mabdinur) [inline ddtrace/contrib/pymongo/patch.py:73]: add `from ddtrace.ext import mongo` to the top of the file

```suggestion
        span.set_tag_str(db.SYSTEM, mongo.SERVICE)
```

**Review by mabdinur — COMMENTED** [review]
some minor nits, overall LGTM

**Review by Yun-Kim — APPROVED** [review]

**Review by mabdinur — APPROVED** [review]

**Pull request merged**